french_law/python/src/allocations_familiales.py

```diff
@@ -545,19 +545,19 @@ def smic(smic_in:SmicIn):
     date_courante = smic_in.date_courante_in
     residence = smic_in.residence_in
     try:
-        def temp_brut_horaire(_:Any):
+        def temp_brut_horaire(_:Unit):
             raise EmptyError
-        def temp_brut_horaire_1(_:Any):
+        def temp_brut_horaire_1(_:Unit):
```

Comment:
What we see in those diffs is that the types that get no constraints attached to them during unification (e.g. arguments of thunks) now get resolved to `Any`,whereas before I used to pick `Unit` for those since we can pick any type. I would prefer the old behavior to be reverted (as well as for the OCaml backend but it's the same problem).

Reply:
The change is in the other way, a quirk made it remain `Any` and I fixed it to be restricted to `Unit` :)